crates/sui/src/ptb/ptb_builder/build_ptb.rs

```diff
@@ -730,35 +762,24 @@ impl<'a> PTBBuilder<'a> {
                 self.resolve(span(arg_loc, PTBArg::Identifier(i)), ctx)
                     .await
             }
-            PTBArg::Bool(b) => ctx.pure(self, arg_loc, b).await,
-            PTBArg::U8(u) => ctx.pure(self, arg_loc, u).await,
-            PTBArg::U16(u) => ctx.pure(self, arg_loc, u).await,
-            PTBArg::U32(u) => ctx.pure(self, arg_loc, u).await,
-            PTBArg::U64(u) => ctx.pure(self, arg_loc, u).await,
-            PTBArg::U128(u) => ctx.pure(self, arg_loc, u).await,
-            PTBArg::U256(u) => ctx.pure(self, arg_loc, u).await,
-            PTBArg::String(s) => ctx.pure(self, arg_loc, s).await,
             x @ PTBArg::Option(_) => {
-                ctx.pure(
-                    self,
-                    arg_loc,
-                    x.into_move_value_opt().map_err(|e| err!(arg_loc, "{e}"))?,
-                )
-                .await
+                ctx.pure(self, arg_loc, x.into_move_value_opt(arg_loc)?)
+                    .await
             }
             x @ PTBArg::Vector(_) => {
-                ctx.pure(
-                    self,
-                    arg_loc,
-                    x.into_move_value_opt().map_err(|e| err!(arg_loc, "{e}"))?,
-                )
-                .await
+                ctx.pure(self, arg_loc, x.into_move_value_opt(arg_loc)?)
+                    .await
             }
             PTBArg::Address(addr) => {
                 let object_id = ObjectID::from_address(addr.into_inner());
                 ctx.resolve_object_id(self, arg_loc, object_id).await
             }
             PTBArg::VariableAccess(head, fields) => {
+                // Since keystore mnemonics can contain dots, we need to resolve these/disambiguate
```

Comment:
I am not sure `keystore mnemonics` can contain dots. I think only aliases can contain dots.

Reply:
Good spot! Mixed the two up in my mind...will fix!